compiler/rustc_middle/src/ty/layout.rs

```diff
@@ -2533,6 +2543,22 @@ where
                     }
                 }
 
+                ty::Dynamic(_, _, ty::DynStar) => {
+                    if i == 0 {
+                        TyMaybeWithLayout::Ty(tcx.types.usize)
+                    } else if i == 1 {
+                        // FIXME(dyn-star) same FIXME as above applies here too
+                        TyMaybeWithLayout::Ty(
+                            tcx.mk_imm_ref(
+                                tcx.lifetimes.re_static,
+                                tcx.mk_array(tcx.types.usize, 3),
+                            ),
+                        )
```

Comment:
Is there some kind of implicit invariant that this must be consistent with `layout_of_uncached`? Nowadays these two functions are not even in the same crate. That seems very fragile. Am I missing something?

Cc @oli-obk

Reply:
I'm trying to fix this as part of https://github.com/rust-lang/rust/pull/107728 but that leads to invalid LLVM IR errors...